app/editor/src/hooks/api-editor/useApi.ts

```diff
@@ -20,42 +19,34 @@ export const useApi = (
   const summon = useSummon({ ...options, baseURL: options.baseURL ?? Settings.ApiPath });
   const [, app] = useAppStore();
 
-  const handleRequest = React.useCallback(
-    async (url: string = 'NO-URL', request: () => Promise<any>) => {
-      try {
-        app.addRequest(url);
-        const response = await request();
-        return response;
-      } catch (error) {
-        // TODO: Capture error information.
-        throw error;
-      } finally {
-        app.removeRequest(url);
-      }
-    },
-    [app],
-  );
+  const handleRequest = async (url: string = 'NO-URL', request: () => Promise<any>) => {
+    try {
+      app.addRequest(url);
+      const response = await request();
+      return response;
+    } catch (error) {
+      // TODO: Capture error information.
+      throw error;
+    } finally {
+      app.removeRequest(url);
+    }
+  };
 
-  const hook = React.useMemo(
-    () =>
-      ({
-        request: (config: AxiosRequestConfig<any>) => {
-          return handleRequest(config.url, () => summon.request(config));
-        },
-        get: (url: string, config?: AxiosRequestConfig<any> | undefined) => {
-          return handleRequest(url, () => summon.get(url, config));
-        },
-        post: (url: string, data?: any, config?: AxiosRequestConfig<any> | undefined) => {
-          return handleRequest(url, () => summon.post(url, data, config));
-        },
-        put: (url: string, data?: any, config?: AxiosRequestConfig<any> | undefined) => {
-          return handleRequest(url, () => summon.put(url, data, config));
-        },
-        delete: (url: string, config?: AxiosRequestConfig<any> | undefined) => {
-          return handleRequest(url, () => summon.delete(url, config));
-        },
-      } as AxiosInstance),
-    [handleRequest, summon],
-  );
-  return hook;
+  return {
```

Comment:
Just returning objects now.  Read up on `useRef` and `useMemo` and `useCallback`.  We're just bloating up memory for simple object creation.